lib/Agrammon/DB/Dataset.pm6

```diff
@@ -27,13 +27,75 @@ class Agrammon::DB::Dataset does Agrammon::DB {
                 RETURNING dataset_id, dataset_mod_date
             DATASET
 
-            my @d = $ds.array;    
+            my @d = $ds.array;
             $!id = @d[0];
             $!mod-date = @d[1];
         }
         return self;
     }
 
+    method rename($new) {
+        self.with-db: -> $db {
+            my $ret = $db.query(q:to/DATASET/, $new, $!name, $!user.id);
+                UPDATE dataset SET dataset_name = $1
+                 WHERE dataset_name = $2 AND dataset_pers = $3
+                RETURNING dataset_name
+            DATASET
+            $!name = $new;
+        }
+        return self;
+    }
+
+    method !clone($new) {
+        my $dataset = Agrammon::DB::Dataset.new(:name($new));
+        warn "clone($new) not yet implemented";
+        return $dataset;
+    }
+
+    method submit($email) {
+        my $new-dataset = self!clone("Clone of $!name");
+
+        warn "Sending mail for submit($email) not yet implemented";
```

Comment:
Sending email is probably not best done in a `DB::Dataset` class anyway. :-)

Reply:
True.